protocols/enocean/org.eclipse.smarthome.protocols.enocean.basedriver.impl/src/main/java/org/eclipse/smarthome/protocols/enocean/basedriver/impl/impl/EnOceanBaseDriver.java

```diff
@@ -0,0 +1,628 @@
+/*******************************************************************************
+ * Copyright (c) 2013, 2015 Orange.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Victor PERRON, Antonin CHAZALET, Andre BOTTARO.
+ *******************************************************************************/
+
+package org.eclipse.smarthome.protocols.enocean.basedriver.impl.impl;
+
+import java.util.Hashtable;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Properties;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Constants;
+import org.osgi.framework.Filter;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.enocean.EnOceanDevice;
+import org.osgi.service.enocean.EnOceanEvent;
+import org.osgi.service.enocean.EnOceanHost;
+import org.osgi.service.enocean.EnOceanMessage;
+import org.osgi.service.enocean.descriptions.EnOceanChannelDescriptionSet;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventAdmin;
+import org.osgi.service.event.EventHandler;
+import org.osgi.util.tracker.ServiceTracker;
+import org.osgi.util.tracker.ServiceTrackerCustomizer;
+
+import org.eclipse.smarthome.protocols.enocean.basedriver.impl.esp.EspPacket;
+import org.eclipse.smarthome.protocols.enocean.basedriver.impl.impl.EnOceanDeviceImpl;
+import org.eclipse.smarthome.protocols.enocean.basedriver.impl.impl.EnOceanHostImpl;
+import org.eclipse.smarthome.protocols.enocean.basedriver.impl.impl.EnOceanHostSerialImpl;
+import org.eclipse.smarthome.protocols.enocean.basedriver.impl.impl.EnOceanHostTestImpl;
+import org.eclipse.smarthome.protocols.enocean.basedriver.impl.radio.Message;
+import org.eclipse.smarthome.protocols.enocean.basedriver.impl.radio.Message1BS;
+import org.eclipse.smarthome.protocols.enocean.basedriver.impl.radio.Message4BS;
+import org.eclipse.smarthome.protocols.enocean.basedriver.impl.radio.MessageRPS;
+import org.eclipse.smarthome.protocols.enocean.basedriver.impl.radio.MessageUTE;
+import org.eclipse.smarthome.protocols.enocean.basedriver.impl.utils.EnOceanHostImplException;
+import org.eclipse.smarthome.protocols.enocean.basedriver.impl.utils.Logger;
+import org.eclipse.smarthome.protocols.enocean.basedriver.impl.utils.Utils;
+
+/**
+ * EnOceanBaseDriver.
+ */
+public class EnOceanBaseDriver implements EnOceanPacketListener,
+		ServiceTrackerCustomizer, EventHandler {
+
+	/** TAG */
+	public static final String TAG = EnOceanBaseDriver.class.getName();
+
+	private BundleContext bc;
+	private Hashtable eoDevices;
```

Comment:
Hi Markus,

No parametrization, we are on CDC 1.1.

Regards,

Antonin.

De : Markus Rathgeb [mailto:notifications@github.com]
Envoyé : vendredi 6 mars 2015 09:14
À : eclipse/smarthome
Cc : CHAZALET Antonin OAB
Objet : Re: [smarthome] EnOcean OSGi Base Driver (https://bugs.eclipse.org/bugs/show_bug.cgi?id=457338) (#155)


In protocols/enocean/org.eclipse.smarthome.protocols.enocean.basedriver.impl/src/main/java/org/eclipse/smarthome/protocols/enocean/basedriver/impl/impl/EnOceanBaseDriver.java<https://github.com/eclipse/smarthome/pull/155#discussion_r25930619>:

> +import org.eclipse.smarthome.protocols.enocean.basedriver.impl.radio.MessageUTE;

> +import org.eclipse.smarthome.protocols.enocean.basedriver.impl.utils.EnOceanHostImplException;

> +import org.eclipse.smarthome.protocols.enocean.basedriver.impl.utils.Logger;

> +import org.eclipse.smarthome.protocols.enocean.basedriver.impl.utils.Utils;

> +

> +/**

> + * EnOceanBaseDriver.

> + */

> +public class EnOceanBaseDriver implements EnOceanPacketListener,

> +           ServiceTrackerCustomizer, EventHandler {

> +

> +    /** TAG */

> +    public static final String TAG = EnOceanBaseDriver.class.getName();

> +

> +    private BundleContext bc;

> +    private Hashtable eoDevices;

Why do you not parameterize this raw type?

—
Reply to this email directly or view it on GitHub<https://github.com/eclipse/smarthome/pull/155/files#r25930619>.

_________________________________________________________________________________________________________________________

Ce message et ses pieces jointes peuvent contenir des informations confidentielles ou privilegiees et ne doivent donc
pas etre diffuses, exploites ou copies sans autorisation. Si vous avez recu ce message par erreur, veuillez le signaler
a l'expediteur et le detruire ainsi que les pieces jointes. Les messages electroniques etant susceptibles d'alteration,
Orange decline toute responsabilite si ce message a ete altere, deforme ou falsifie. Merci.

This message and its attachments may contain confidential or privileged information that may be protected by law;
they should not be distributed, used or copied without authorisation.
If you have received this email in error, please notify the sender and delete this message and its attachments.
As emails may be altered, Orange is not liable for messages that have been modified, changed or falsified.
Thank you.